src/lib.rs

```diff
@@ -77,18 +79,26 @@ impl MsgEnvelope {
     /// So, needs some improvement..
```

Comment:
better use `TODO` and be more specific about what need to improve?

Reply:
yeah, agreed!